furms-core/src/main/java/io/imunity/furms/core/config/security/method/FurmsMethodSecurityExpressionRoot.java

```diff
@@ -29,12 +30,20 @@ public boolean hasCapabilityForResource(Capability capability, ResourceType reso
 	}
 
 	public boolean hasCapabilityForResource(Capability capability, ResourceType resourceType, String id) {
-		FurmsAuthenticatedUser principal = (FurmsAuthenticatedUser)authentication.getPrincipal();
-		ResourceId resourceId = new ResourceId(id, resourceType);
+		if(!authentication.isAuthenticated() || hasAnonymousRole())
+			return false;
 
+		FurmsAuthenticatedUser principal = (FurmsAuthenticatedUser) authentication.getPrincipal();
+		ResourceId resourceId = new ResourceId(id, resourceType);
 		return capabilityCollector.getCapabilities(principal.getRoles(), resourceId).contains(capability);
 	}
 
+	private boolean hasAnonymousRole() {
+		return authentication.getAuthorities().stream()
+			.map(GrantedAuthority::getAuthority)
+			.anyMatch(role -> role.equals("ROLE_ANONYMOUS"));
```

Comment:
2 notes here: is there a constant in Spring for that? 
shouldn't it be checked in the first place whether authentication has set the "authenticated" flag?